DOCKER_README.MD

```diff
@@ -0,0 +1,88 @@
+This guide shows our recommended recommended way of installing Docker on your OS X machine. 
+
+## Install docker
+
+First install Docker using [Homebrew](https://brew.sh/)
+```
+$ brew install docker
+```
+
+You can install [Docker Desktop](https://docs.docker.com/get-docker/) if you wish, or use the `docker-machine` command to work with your Docker containers. This guide assumes you use the `docker-machine` command.
+
+## Setup your Docker
+
+Install `docker-machine`
+```
+$ brew install docker-machine
+```
+
+Then install [VirtualBox](https://www.virtualbox.org/) using [Homebrew Cask](https://github.com/Homebrew/homebrew-cask) to get a driver for your Docker machine
+```
+$ brew cask install virtualbox
+```
+
+You may need to enter your password and authorize the application through `System Settings` > `Security & Privacy`.
+
+Create a new machine, and set it up as default, and connect your shell to the machine with the following commands:
+
+```
+$ docker-machine create --driver virtualbox default
+$ docker-machine env default
+$ eval "$(docker-machine env default)"
+```
+
+Now you're all setup to use our provided Docker image!
+
+## Build the image
+
+```bash
+docker build -t algolia-ruby .
+```
+
+## Run the image
+
+You need to provide few environment variables at runtime to be able to run the [Common Test Suite](https://github.com/algolia/algoliasearch-client-specs/tree/master/common-test-suite).
+You can set them up directly in the command
+
+```bash
+docker run -it --rm --env ALGOLIA_APPLICATION_ID_1=XXXXXX [...] -v $PWD:/app -w /app algolia-ruby bash
+```
+
+But we advise you export them in your `.bashrc` or `.zshrc`. That way, you can use [Docker's shorten syntax](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) to retrieve your variables.
+
+```bash
+### For external contributors, only the following env variables should be enough
+docker run -it --rm --env ALGOLIA_APPLICATION_ID_1 \
```

Comment:
Maybe we should format the commands like this, so that it's easier to copy and edit it before running the command. WDYT?

We would need to do this for all the environment variables in that case
```suggestion
docker run -it --rm --env ALGOLIA_APPLICATION_ID_1="your_app_id" \
```

Reply:
@DevinCodes If you have exported the env variable with that name it doesn't need to be defined, Docker will go and fetch it by itself. This was noted a bit above so do you think it needs to be reworded for clarity?

Reply:
Sorry my bad, I guess I went over that part too fast during the review then 😓